### PR TITLE
Enable custom service name for Android sourcemaps

### PR DESCRIPTION
**Pull request by louiszawadzki**
### What does this PR do?

Enable custom service name for Android sourcemaps.

### Motivation

Enable customers using a custom service name to upload their sourcemaps automatically on Android.
The solution does not require to add new files, as this is not so common.

### Additional Notes

If users need to specify different service name per environment, they can use environment variables.

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)
- [x] If this PR is auto-generated, please make sure also to manually update the code related to the change

### CODE REVIEW

File: docs/crash_reporting.md [hunk 1]
```diff
@@ -66,14 +66,30 @@ To set your project up to send the symbolication files automatically, run `npx d
...
+
+```groovy
+project.ext.datadog = [
+    serviceName: "com.my.custom.service"
+]
+```
+
+### On iOS using the `datadog-ci react-native xcode` command
+
+Available options for the `datadog-ci react-native xcode` command are available on the [command documentation page].
```
Comment (fuzzybinary) [inline docs/crash_reporting.md:83]: Shole "command documentation page" have a footnote link?

**Review by fuzzybinary — APPROVED** [review]

File: docs/crash_reporting.md (hunk 1 above)
Comment (rtrieu) [reply docs/crash_reporting.md:83]: +1

**Review by rtrieu — CHANGES_REQUESTED** [review]
Missing link

File: packages/core/datadog-sourcemaps.gradle [hunk 2]
```diff
@@ -189,6 +189,11 @@ private def runShellCommand(
     }
 }
 
-private static def getServiceName(variant) {
+private def getServiceName(variant) {
+    if (project.hasProperty("datadog")) {
```
Comment (0xnm) [inline packages/core/datadog-sourcemaps.gradle:193]: won't something like this work? anyway, this doesn't really change anything.

```suggestion
    if (project.datadog != null && project.datadog.serviceName != null) {
```

**Review by 0xnm — APPROVED** [review]

File: docs/crash_reporting.md [hunk 3]
```diff
@@ -66,14 +66,30 @@ To set your project up to send the symbolication files automatically, run `npx d
...
+
+```groovy
+project.ext.datadog = [
+    serviceName: "com.my.custom.service"
+]
+```
+
+### On iOS using the `datadog-ci react-native xcode` command
+
+Available options for the `datadog-ci react-native xcode` command are available on the [command documentation page](12).
```
Comment (rtrieu) [inline docs/crash_reporting.md:83]:
```suggestion
Available options for the `datadog-ci react-native xcode` command are available on the [command documentation page][12].
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Minor update - link formatting.

**Review by rtrieu — APPROVED** [review]
Looks good!

**Pull request merged**